generate_aliases/pkg/generate/generate.go

```diff
@@ -97,50 +84,45 @@ func Generate(filePath, targetPath string) error {
 }
 
 // GenerateAlias -- TODO document
-func GenerateAlias(w io.Writer, aliasNames map[string]string, alias *types.AliasCMD) error {
+func GenerateAlias(w io.Writer, aliases *types.AliasCMDs) error {
+	for _, alias := range aliases.Aliases {
+		if alias.Prefix != "" {
+			alias.Prefix = strings.Trim(alias.Prefix, " ") + " "
+		}
+		if alias.Suffix != "" {
+			alias.Suffix = " " + strings.Trim(alias.Suffix, " ")
+		}
 
-	if alias.Prefix != "" {
-		alias.Prefix = strings.Trim(alias.Prefix, " ") + " "
-	}
-	if alias.Suffix != "" {
-		alias.Suffix = " " + strings.Trim(alias.Suffix, " ")
-	}
+		aliasName := fmt.Sprintf("%vk%v%v%v",
+			alias.PrefixShort,
+			alias.Short,
+			alias.ResourceShort,
+			alias.SuffixShort)
+		aliasCommand := fmt.Sprintf("%vkubectl %v %v%v",
+			alias.Prefix,
+			alias.CMD,
+			alias.Resource,
+			alias.Suffix)
+
+		if v, ok := aliases.AliasNames[aliasName]; ok {
```

Comment:
Totally optional:

I can create a function to add aliases to AliasCMDs, and then when we add them we can do this whole existence checking code block. That way `AliasNames` never gets exposed in the code.

I don't need it, but I realized I may have only half solved the leaky abstraction :D

Reply:
Works for me.